src/app/dataValidator/rules/topstories.js

```diff
@@ -0,0 +1,55 @@
+import deepClone from '../../helpers/json/deepClone';
+import deepGet from '../../helpers/json/deepGet';
+
+const squashKeys = [
+  'responsive-top-stories',
+  'top-story',
+  'secondary-top-story',
+  'other-top-stories',
+];
+
+const squashTopStories = jsonRaw => {
+  const json = deepClone(jsonRaw); // make a copy so we don't corrupt the input
+  let groups = deepGet(['content', 'groups'], json);
+  let collectedItems = [];
+
+  if (groups) {
+    /*
+     * Find and delete unwanted groups while collecting items
+     */
+    groups = groups.filter(group => {
+      if (group.type && squashKeys.includes(group.type)) {
+        collectedItems = collectedItems.concat(group.items);
+
+        return false; // delete the group
+      }
+
+      return true; // keep the group
+    });
+
+    /*
+     * If items were collected, form into a new group
+     */
+    if (collectedItems.length > 0) {
+      const newTopStoriesGroup = {
+        type: 'top-stories',
+        title: 'Top stories',
+        items: collectedItems,
+      };
+
+      /*
+       * Add new group to the start of the groups array
+       */
+      groups.unshift(newTopStoriesGroup);
+
+      /*
+       * Override the original groups with the new ones
```

Comment:
We make a [deep clone](https://github.com/bbc/simorgh/pull/1596/files#diff-647ce4d4a42e13902b4fdc37c884a0bfR12) not to change the original input but then overwrite the groups anyway. Considering the deepClone needs to process on the client and then we replace it anyway I'm not convinced the benefit of the cloning is worthwhile - unless I'm missing something.

Reply:
If we dont deep clone we manipulate function arguments, which I dont think is good practice

Reply:
Not convinced but would like to hear other peoples opinions. If we never compare the original input to the manipulated output I don't see why we should clone/maintain the original input

Reply:
I agree with Phil. If we're not really using the original data, then does it make sense to clone it?